daemons/controld/controld_based.c

```diff
@@ -43,7 +45,11 @@ do_cib_replaced(const char *event, xmlNode * msg)
 
     /* start the join process again so we get everyone's LRM status */
     populate_cib_nodes(node_update_quick|node_update_all, __func__);
-    register_fsa_input(C_FSA_INTERNAL, I_ELECTION, NULL);
+
+    crm_element_value_int(msg, F_CIB_CHANGE_SECTION, &change_section);
+    if (change_section & (cib_change_section_nodes | cib_change_section_status)) {
+        register_fsa_input(C_FSA_INTERNAL, I_ELECTION, NULL);
+    }
```

Comment:
What I meant about populate_cib_nodes() earlier was that we can simply move the populate_cib_nodes() call above inside the "if" block, since it only needs to be called if the nodes changed. That would help avoid multiple asynchronous CIB queries (inside populate_cib_nodes()) when unnecessary.

Reply:
I intended to change populate_cib_nodes () in the next PR.
I intended to make the following corrections, but what do you think?

```
    crm_element_value_int(msg, F_CIB_CHANGE_SECTION, &change_section);
    if (change_section & (cib_change_section_nodes | cib_change_section_status)) {
        
        register_fsa_input(C_FSA_INTERNAL, I_ELECTION, NULL);
    }
```

Reply:
Yes, that looks good. Either another PR or another commit in this PR is fine

Reply:
Okay!